pkg/vsphere/tasks/waiter.go

```diff
@@ -63,22 +62,45 @@ func Wait(ctx context.Context, f func(context.Context) (Waiter, error)) error {
 //       return vm.Reconfigure(ctx, config)
 //    })
 func WaitForResult(ctx context.Context, f func(context.Context) (ResultWaiter, error)) (*types.TaskInfo, error) {
-	task, err := f(ctx)
+	r := rand.New(rand.NewSource(time.Now().UnixNano())) //creates a more unique random
+	var err error
+	var taskInfo *types.TaskInfo
+	backoffFactor := int64(1)
+
+	t, err := f(ctx)
 	if err != nil {
-		cerr := errors.Errorf("Failed to invoke operation: %s", errors.ErrorStack(err))
-		log.Errorf(cerr.Error())
-		return nil, cerr
+		log.Error(err)
+		return nil, err
 	}
 
-	info, err := task.WaitForResult(ctx, nil)
-	if err != nil {
-		cerr := errors.Errorf("Operation failed: %s", errors.ErrorStack(err))
-		if info != nil && info.Error != nil {
-			cerr = errors.Errorf("%s - (%s)", cerr, info.Error)
+	for {
+		taskInfo, err := t.WaitForResult(ctx, nil)
+
+		if err == nil {
+			return taskInfo, nil
 		}
 
-		log.Errorf(cerr.Error())
-		return nil, cerr
+		if terr, ok := err.(*task.Error); ok {
+			if _, ok := terr.Fault().(*types.TaskInProgress); ok {
+				sleepValue := time.Duration(backoffFactor * (r.Int63n(100) + int64(50)))
+				select {
+				case <-time.After(sleepValue * time.Millisecond):
+					if backoffFactor*2 > maxBackoffFactor {
+						backoffFactor = maxBackoffFactor
+					} else {
+						backoffFactor *= 2
+					}
+				case <-ctx.Done():
+					log.Errorf("Context Deadline Exceeded while trying to Retry task : %#v", taskInfo)
+					return nil, ctx.Err()
+				}
+				log.Infof("Retrying Task due to TaskInProgressFault: %s", taskInfo.Task.Reference())
```

Comment:
log.Warnf instead?  Rather than "TaskInProgressFault" literal, you could %#v terr.Fault(), that'll include the fault type name and the moref of the task that is currently in progress.